gloo/transport/tcp/tls/openssl/openssl.cc

```diff
@@ -0,0 +1,111 @@
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+
+#include "dynamic_library.h"
+
+namespace {
+DynamicLibrary libssl("libssl.so", "libssl.so.1.1");
+} // namespace
+
+
+#define CALL_SYM(NAME, ...) \
+static auto fn = reinterpret_cast<decltype(&NAME)>(libssl.sym(__func__)); \
+return fn(__VA_ARGS__)
+
+void ERR_print_errors_cb(int (*cb) (const char *, size_t, void *), void *u) {
+  CALL_SYM(ERR_print_errors_cb, cb, u);
```

Comment:
To avoid symbols clashes with the actual libssl, can those function be defined in their own namespace? Say `_glootls::`?

Reply:
addressed